src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5842,6 +5842,17 @@ private BoundExpression BindCheckedExpression(CheckedExpressionSyntax node, Diag
         /// cannot.  If this is not a Color Color case, then return null.
         /// </summary>
         private BoundExpression BindLeftOfPotentialColorColorMemberAccess(ExpressionSyntax left, DiagnosticBag diagnostics)
+        {
+            if (left is IdentifierNameSyntax identifier)
+            {
+                return BindLeftIdentifierOfPotentialColorColorMemberAccess(identifier, diagnostics);
+            }
+
+            // NOTE: it is up to the caller to call CheckValue on the result.
+            return BindExpression(left, diagnostics);
+        }
+
+        private BoundExpression BindLeftIdentifierOfPotentialColorColorMemberAccess(IdentifierNameSyntax left, DiagnosticBag diagnostics)
```

Comment:
Should we mark this to prevent inlining as that would deafeat the purpose of this change